internal/api/initialize/init_lokiStack.go

```diff
@@ -6,47 +6,86 @@ import (
 
 	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
 	internalobs "github.com/openshift/cluster-logging-operator/internal/api/observability"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
 	lokioutput "github.com/openshift/cluster-logging-operator/internal/generator/vector/output/loki"
 	"github.com/openshift/cluster-logging-operator/internal/utils"
 )
 
-// MigrateLokiStack migrates a lokistack output into appropriate loki outputs based on defined inputs
+const (
+	lokiOtlpEndpoint = "/otlp/v1/logs"
+)
+
+// MigrateLokiStack migrates a lokistack output into appropriate loki or OTLP outputs based on defined inputs
 func MigrateLokiStack(spec obs.ClusterLogForwarder, options utils.Options) obs.ClusterLogForwarder {
 	var outputs []obs.OutputSpec
 	var pipelines []obs.PipelineSpec
-	outputs, pipelines = ProcessForwarderPipelines(spec.Spec, obs.OutputTypeLokiStack, "", false)
+
+	outputs, pipelines = ProcessForwarderPipelines(spec.Spec)
 
 	spec.Spec.Outputs = outputs
 	spec.Spec.Pipelines = pipelines
 
 	return spec
 }
 
-func GenerateLokiOutput(outSpec obs.OutputSpec, input, tenant string) obs.OutputSpec {
-	return obs.OutputSpec{
-		Name: fmt.Sprintf("%s-%s", outSpec.Name, input),
-		Type: obs.OutputTypeLoki,
-		Loki: &obs.Loki{
-			URLSpec: obs.URLSpec{
-				URL: lokiStackURL(outSpec.LokiStack, tenant),
-			},
-			Authentication: &obs.HTTPAuthentication{
-				Token: outSpec.LokiStack.Authentication.Token,
-			},
-			Tuning:    outSpec.LokiStack.Tuning,
-			LabelKeys: lokiStackLabelKeysForTenant(outSpec.LokiStack.LabelKeys, tenant, lokioutput.DefaultLabelKeys),
-		},
+// GenerateOutput returns either a Loki or OTLP output spec when migrating Lokistacks
+func GenerateOutput(outSpec obs.OutputSpec, input, tenant string) obs.OutputSpec {
+	obsOut := &obs.OutputSpec{
+		Name:  fmt.Sprintf("%s-%s", outSpec.Name, input),
 		TLS:   outSpec.TLS,
 		Limit: outSpec.Limit,
 	}
+
+	if outSpec.LokiStack.DataModel == constants.DataModelOtel {
+		obsOut.Type = obs.OutputTypeOTLP
+		obsOut.OTLP = GenerateOtlpSpec(outSpec.LokiStack, tenant)
+	} else {
+		obsOut.Type = obs.OutputTypeLoki
+		obsOut.Loki = GenerateLokiSpec(outSpec.LokiStack, tenant)
+	}
+
+	return *obsOut
+}
+
+// GenerateLokiSpec generates and returns a Loki spec for the defined lokistack output
+func GenerateLokiSpec(ls *obs.LokiStack, tenant string) *obs.Loki {
+	return &obs.Loki{
+		URLSpec: obs.URLSpec{
+			URL: lokiStackURL(ls, tenant, false),
+		},
+		Authentication: &obs.HTTPAuthentication{
+			Token: ls.Authentication.Token,
+		},
+		Tuning:    ls.Tuning,
+		LabelKeys: lokiStackLabelKeysForTenant(ls.LabelKeys, tenant, lokioutput.DefaultLabelKeys),
+	}
 }
 
-func lokiStackURL(lokiStackSpec *obs.LokiStack, tenant string) string {
+// GenerateOtlpSpec generates and returns an OTLP spec for the defined lokistack output
+// Note: OTLP does not support compression type `snappy` where loki does
```

Comment:
Loki using the current ingestion point supports snappy which AFAIK is just a plain HTTP output (@xperimental?).  The restriction as it exists now is in the mark up validation which occurs before any of our validation code so it shouldn't be a hard blocker.  I presume the reason this may not work as-is because "snappy" is not a compression enum value for OTLP, otherwise I would not expect any other restrictions.  Maybe this is a place where an "internal" model should exist which would allow us to pass on this value as something other than an enum

Reply:
Maybe this is a question for @cahartma? What was the decision for only allowing `none` and `gzip` when the `HTTP` output defines these as the choices: `none;gzip;snappy;zlib`?